tests/Dfc.CourseDirectory.WebV2.Tests/FeatureTests/Providers/ProviderDetailsTests.cs

```diff
@@ -90,6 +92,45 @@ public async Task Get_ValidRequest_RendersExpectedContent(TestUserType userType)
             Assert.Equal("F.E.", doc.GetSummaryListValueWithKey("Provider type"));
         }
 
+        [Theory]
+        [InlineData(TestUserType.Developer)]
+        [InlineData(TestUserType.Helpdesk)]
+        [InlineData(TestUserType.ProviderSuperUser)]
+        [InlineData(TestUserType.ProviderUser)]
+        public async Task Get_ValidRequestWithTLevelsProviderWithSelectedTLevelsDefinitions_RendersExpectedContent(TestUserType userType)
+        {
+            var providerId = await TestData.CreateProvider(providerType: ProviderType.TLevels);
+
+            var tLevelDefinitionIds = await Task.WhenAll(Enumerable.Range(0, 3).Select(_ => Guid.NewGuid()).Select(id => TestData.CreateTLevelDefinition(tLevelDefinitionId: id, name: $"Name-{id}")));
+            var selectedTLevelDefinitionIds = tLevelDefinitionIds.OrderBy(_ => Guid.NewGuid()).Take(2).ToArray();
+            await TestData.SetProviderTLevelDefinitions(providerId, selectedTLevelDefinitionIds);
+
+            var request = new HttpRequestMessage(HttpMethod.Get, $"providers?providerId={providerId}");
+
+            await User.AsTestUser(userType, providerId);
+
+            // Act
+            var response = await HttpClient.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            var doc = await response.GetDocument();
+            var providerTypeValue = doc.GetSummaryListValueWithKey("Provider type");
+
+            providerTypeValue.Should().Contain("T Levels");
```

Comment:
If you know a better way to assert this I'd love to hear it.

Reply:
Perhaps a similar method to `GetSummaryListValueWithKey` that returns the `IElement` instead of its `TextContent` then could loop over the `<li>`s

Reply:
`GetSummaryListValueElementWithKey`?

Reply:
That perhaps would work but it were not for the way in which we're displaying the list of TLevelDefinitions (using a govuk-details component) - going down this route would make the test considerably more fragile and too closely tied to the implementation IMO.

Reply:
Could put a `data-testid` attribute on the component and just check if it exists?